src/rust/engine/src/rule_graph.rs

```diff
@@ -794,6 +849,35 @@ impl RuleEdges {
     }
     RuleEdges { dependencies: new_deps, selector_to_dependencies: new_selector_deps }
   }
+
+  pub fn entries_for(&self, selectors: &Vec<Selector>) -> Entries {
+    let r = self.selector_to_dependencies.get(selectors).map(|x|x.clone()).unwrap_or_else(|| Vec::new());
+
+    if false {
```

Comment:
This is going to be a bit too noisy to keep in, but for future reference:
```
`externs::log(LogLevel, "msg")`
```
... exists.